community/modules/compute/gke-node-pool/main.tf

```diff
@@ -158,6 +169,12 @@ resource "google_container_node_pool" "node_pool" {
       condition     = !(var.local_ssd_count_ephemeral_storage > 0 && var.local_ssd_count_nvme_block > 0)
       error_message = "Only one of local_ssd_count_ephemeral_storage or local_ssd_count_nvme_block can be set to a non-zero value."
     }
+
+    # non preattached gpu machine type should always have guest_accelerator config defined
+    precondition {
+      condition     = (local.preattached_gpu_machine_family && length(var.guest_accelerator != null ? var.guest_accelerator : []) >= 0) || (!local.preattached_gpu_machine_family && length(var.guest_accelerator != null ? var.guest_accelerator : []) > 0)
```

Comment:
This condition does not make sense to me: `length(var.guest_accelerator != null ? var.guest_accelerator : []) >= 0)`, it seems always true. 

var.guest_accelerator = null
var.guest_accelerator = []
var.guest_accelerator = [item]

all true.